### DISCUSSION TITLE
v4 migration

### DISCUSSION THREAD

**Discussion by drph4nt0m** [drph4nt0m/avbot#197] [Ideas]
- Migrate to ts using https://discord-ts.js.org/ and deprecate simple commands (only use slash commands).
- Use Node.js v16.13.x
- MongoDB not required in initial upgrade

# Commands

## HIGH priority

- [x] `weather/atis`
- [x] `weather/brief`
- [x] `weather/metar`
- [x] `weather/nats`
- [x] `weather/raw-metar` (merge into the `weather/metar` command/slash group with `parsed` and `raw` as sub commands. parsed wont need to send raw)
- [x] `weather/raw-taf` (merge into the `weather/taf` command/slash group with `parsed` and `raw` as sub commands. parsed wont need to send raw))
- [x] `weather/taf`

## Normal Priority
- [x] `ivao/ivao-online`
- [x] `ivao/ivao`
- [x] `vatsim/vatsim-online`
- [x] `vatsim/vatsim`
- [x] `misc/icao`
- [x] `misc/search`
- [x] `time/local`
- [x] `time/zulu`

## Low Priority
- [x] `util/donate` (merge `util/donate` and `util/invite` into a `util/info` command with some other extra info like uptime, owner, support server link, etc)
- [x] `util/help`
- [x] `util/invite`
- [x] `util/ping` (remove the restriction, anyone can use it)
- [x] `weather/atis-voice` (merge into the `weather/atis` command along with a option for voice)

## Not needed in initial upgrade
- [ ] `misc/chart`
- [x] `misc/live`
- [ ] `util/api`

## Depricated
- [ ] `util/prefix`
- [ ] `util/stat`
- [ ] `util/top`

**Comment by drph4nt0m** [comment]
cc @VictoriqueMoe

**Comment by VictoriqueMoe** [comment]
`metar` is done. code architecture is written, upsert for Node, DiscordTS and slash commmands are done, DI system is in place and bot is running, command such as `metar` are executing as of right now.

**Comment by VictoriqueMoe** [comment]
I am going to get all high priority done today hopefully

**Comment by VictoriqueMoe** [comment]
Command Guards for Premium and rate limiting is also done.  (and so is Mongo)

**Comment by VictoriqueMoe** [comment]
`weather/atis`, `weather/brief`, `weather/metar`, `weather/nats`, `weather/raw-metar`, `weather/raw-taf` and `weather/taf` are all now complete.

**Comment by VictoriqueMoe** [comment]
I have written all normal priority commands and will do all the `Low Priority` today/tomorrow
